lib/regular_expression/bytecode.rb

```diff
@@ -14,7 +14,7 @@ def self.compile(nfa)
 
       visited = Set.new
       worklist = [[nfa, [:jump_to_fail]]]
-
+      n_of_transitions = 0
```

Comment:
Can you move this work into a separate PR? It seems like it would help us in all backends, so we should move this up.